Scripts/Tracking/Collision/ListProcess/BaseListProcess.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace VRTK.Core.Tracking.Collision.ListProcess
+{
+    using UnityEngine;
+    using System.Linq;
+    using System.Collections.Generic;
+    using VRTK.Core.Tracking.Collision.ListProcess.Action;
+    using VRTK.Core.Extension;
+
+    /// <summary>
+    /// The base for providing a processor to a <see cref="CollisionList"/>.
+    /// </summary>
+    public abstract class BaseListProcess : MonoBehaviour
+    {
+        /// <summary>
+        /// Executes any <see cref="BaseListAction"/> components found on the collider.
+        /// </summary>
+        [Tooltip("Executes any List Action components found on the collider.")]
+        public bool executeColliderActions = false;
+
+        /// <summary>
+        /// Emitted when the <see cref="CollisionList"/> collection has been processed.
+        /// </summary>
+        public CollisionList.UnityEvent Processed = new CollisionList.UnityEvent();
+
+        protected List<CollisionNotifier.EventData> processedList = new List<CollisionNotifier.EventData>();

```

Comment:
Is this `previousList`? It seems like not all subclasses use it - should this instead be a thing in the subclasses that need it?